apache/netbeans
# Add initial support for Jakarta 9.1 and improve GlassFish support for all released versions

Changes for NetBeans in general

- Add initial support for Jakarta EE 9.1 (missing schemas)
- Add java 11 validations for Jakarta EE 9.1
- Add some missing Java EE validations
- Remove not working logic to preselect a server
- Use switch-case instead of long if-else statements

Changes only for GlassFish

- Add support for Jakarta EE 9.1
- Add missing support for GlassFish 4 (JEE 7)
- Add support for every GlassFish version released up to 6.2.5
- Add tests for GlassFish automatic registration
- Tailored config files for every GlassFish version (I downloaded every GlassFish version and check the libraries and release notes)
  - Included libraries
  - Javadoc links
  - Supported java se versions
  - Supported java ee versions

Before - Opening a JakartaEE 9.1 project and NetBeans can not detect the Server and the Java EE version
![Before-JakartaEE91](https://user-images.githubusercontent.com/9832133/178367823-cf1bf9ac-c4e8-4259-b677-c28e6e45a848.png)
After
![After-JakarteEE91](https://user-images.githubusercontent.com/9832133/178367832-a86b697b-9db5-427d-952f-5aef14f44fbd.png)

Testing:
- Full build done
- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Sigtests
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Download every version of GlassFish and successfully:
  - Register it
  - Check correct Java SE support from Java 6 to Java 18
  - Check correct Java/Jakarta EE support from JEE 5 to JakartaEE 9.1
  - Create Web App Project
  - Deploy and run a servlet

[GlassFish GitHub](https://github.com/eclipse-ee4j/glassfish)
